source-mongodb/pull.go

```diff
@@ -140,30 +179,32 @@ type changeEvent struct {
 const resumePointGoneErrorMessage = "the resume point may no longer be in the oplog"
 const resumeTokenNotFoundErrorMessage = "the resume token was not found"
 
-func (c *capture) ChangeStream(ctx context.Context, client *mongo.Client, binding int, res resource, backfillFinishedAt *primitive.Timestamp, resumeToken bson.Raw) error {
+func (c *capture) ChangeStream(ctx context.Context, client *mongo.Client, binding int, res resource, state *resourceState) error {
 	var db = client.Database(res.Database)
 
 	var collection = db.Collection(res.Collection)
 
 	log.WithFields(log.Fields{
 		"database":           res.Database,
 		"collection":         res.Collection,
-		"resumeToken":        resumeToken,
-		"backfillFinishedAt": backfillFinishedAt,
+		"state":              state,
 	}).Info("listening on changes on collection")
 	var eventFilter = bson.D{{"$match", bson.D{{"$or",
-		bson.A{
-			bson.D{{"operationType", "delete"}},
-			bson.D{{"operationType", "insert"}},
-			bson.D{{"operationType", "update"}},
-			bson.D{{"operationType", "replace"}},
-		}}}}}
+	bson.A{
+		bson.D{{"operationType", "delete"}},
+		bson.D{{"operationType", "insert"}},
+		bson.D{{"operationType", "update"}},
+		bson.D{{"operationType", "replace"}},
+	}}}}}
+
 	var opts = options.ChangeStream().SetFullDocument(options.UpdateLookup)
-	if resumeToken != nil {
-		opts = opts.SetResumeAfter(resumeToken)
-	} else if backfillFinishedAt != nil {
-		opts = opts.SetStartAtOperationTime(backfillFinishedAt)
+	if state.StreamResumeToken != nil {
+		opts = opts.SetResumeAfter(state.StreamResumeToken)
+	} else if !state.BackfillStartedAt.IsZero() {
```

Comment:
Will `state.BackfillStartedAt.IsZero` ever be `true`?

Reply:
@williamhbaker it shouldn't be